x-pack/plugins/cases/public/components/user_actions/index.tsx

```diff
@@ -5,251 +5,201 @@
-      box-shadow: none;
-      .euiCommentEvent__header {
-        padding: ${theme.eui.euiSizeM} ${theme.eui.euiSizeS};
-        border-bottom: 0;
-      }
-    }
-  `}
+const BottomUserActionsListWrapper = styled(EuiFlexItem)`
+  padding-top: 16px;
 `;
 
-export const UserActions = React.memo(
-  ({
-    caseConnectors,
-    caseUserActions,
-    userProfiles,
+export const UserActions = React.memo((props: UserActionTreeProps) => {
+  const {
     currentUserProfile,
     data: caseData,
-    getRuleDetailsHref,
-    actionsNavigation,
-    isLoadingUserActions,
-    onRuleDetailsClick,
-    onShowAlertDetails,
-    onUpdateField,
     statusActionButton,
     useFetchAlertData,
-    filterOptions,
-  }: UserActionTreeProps) => {
-    const { detailName: caseId, commentId } = useCaseViewParams();
-    const [initLoading, setInitLoading] = useState(true);
-    const {
-      externalReferenceAttachmentTypeRegistry,
-      persistableStateAttachmentTypeRegistry,
-      appId,
-    } = useCasesContext();
-
-    const alertIdsWithoutRuleInfo = useMemo(
-      () => getManualAlertIdsWithNoRuleId(caseData.comments),
-      [caseData.comments]
-    );
-
-    const [loadingAlertData, manualAlertsData] = useFetchAlertData(alertIdsWithoutRuleInfo);
-
-    const {
-      loadingCommentIds,
-      commentRefs,
-      selectedOutlineCommentId,
-      manageMarkdownEditIds,
-      handleManageMarkdownEditId,
-      handleOutlineComment,
-      handleSaveComment,
-      handleManageQuote,
-      handleDeleteComment,
-      handleUpdate,
-    } = useUserActionsHandler();
-
-    const MarkdownNewComment = useMemo(
-      () => (
-        <AddComment
-          id={NEW_COMMENT_ID}
-          caseId={caseId}
-          ref={(element) => (commentRefs.current[NEW_COMMENT_ID] = element)}
-          onCommentPosted={handleUpdate}
-          onCommentSaving={handleManageMarkdownEditId.bind(null, NEW_COMMENT_ID)}
-          showLoading={false}
-          statusActionButton={statusActionButton}
-        />
-      ),
-      [caseId, handleUpdate, handleManageMarkdownEditId, statusActionButton, commentRefs]
-    );
-
-    useEffect(() => {
-      if (initLoading && !isLoadingUserActions && loadingCommentIds.length === 0) {
-        setInitLoading(false);
-        if (commentId != null) {
-          handleOutlineComment(commentId);
-        }
-      }
-    }, [commentId, initLoading, isLoadingUserActions, loadingCommentIds, handleOutlineComment]);
+    userActivityQueryParams,
+    userActionsStats,
+  } = props;
+  const { detailName: caseId } = useCaseViewParams();
+  const {
+    infiniteCaseUserActions,
+    isLoadingInfiniteUserActions,
+    lastPage,
+    hasNextPage,
+    fetchNextPage,
+    showLoadMore,
+    showBottomList,
+  } = useUserActionsPagination({
+    userActivityQueryParams,
+    userActionsStats,
+    caseId: caseData.id,
+  });
+
+  const { isLoadingLastPageUserActions, lastPageUserActions } = useLastPageUserActions({
```

Comment:
created separate hook for last page user actions for clear separation of concerns as it was confusing for me to have it in pagination hook.